web/src/components/Korrel8rPanel.tsx

```diff
@@ -70,25 +101,94 @@ export default function Korrel8rPanel() {
-          Query
-        </Button>
-      </FlexItem>
+        <Title headingLevel="h3">Correlated Signals</Title>
+        <FlexItem>
+          <ExpandableSectionToggle
+            contentId={queryContentID}
+            toggleId={queryToggleID}
+            isExpanded={showQuery}
+            onToggle={(on: boolean) => setShowQuery(on)}
+          >
+            {showQuery ? 'Hide' : 'Show'} Query
+          </ExpandableSectionToggle>
+        </FlexItem>
+      </Level>
+      <ExpandableSection
+        contentId={queryContentID}
+        toggleId={queryToggleID}
+        isExpanded={showQuery}
+        isDetached
+        isIndented
+      >
+        <Flex direction={{ default: 'column' }}>
+          <Tooltip content={t('Korrel8 query selecting the starting points for correlation.')}>
+            <TextInput
+              className="tp-plugin__panel-query-input"
+              placeholder="domain:class:querydata"
+              id={queryInputID}
+              value={query}
+              onChange={(_event, value) => setQuery(value)}
+            />
+          </Tooltip>
+          <Flex>
+            <Radio
+              label={t('Show neighbourhood of depth')}
+              name={queryTypeOptions}
+              id="neighbourhood-option"
+              isChecked={queryType === QueryType.Neighbour}
+              onChange={(_: React.FormEvent, on: boolean) => {
+                on && setSearchType(QueryType.Neighbour);
+              }}
+            />
+            <FlexItem>
+              <NumberInput
+                value={depth}
+                isDisabled={queryType !== QueryType.Neighbour}
+                onMinus={() => (depth > 1 ? setDepth(depth + 1) : setDepth(1))}
+                onPlus={() => setDepth((depth || 0) + 1)}
+                onChange={(event: React.FormEvent<HTMLInputElement>) => {
+                  const n = Number((event.target as HTMLInputElement).value);
+                  setDepth(isNaN(n) ? 1 : n);
+                }}
+              />
+            </FlexItem>
+          </Flex>
+          <Flex>
+            <Radio
+              label={t('Find paths to goal class')}
+              name={queryTypeOptions}
+              id="goal-option"
+              isChecked={queryType === QueryType.Goal}
+              onChange={(_: React.FormEvent, on: boolean) => on && setSearchType(QueryType.Goal)}
+            />
+            <FlexItem>
+              <TextInput
+                value={goal}
+                isDisabled={queryType !== QueryType.Goal}
+                placeholder="domain:class"
+                onChange={(event: React.FormEvent<HTMLInputElement>) => {
+                  setGoal((event.target as HTMLInputElement).value);
+                }}
+              />
+            </FlexItem>
+            <FlexItem align={{ default: 'alignRight' }}>
+              <Button
+                isAriaDisabled={
+                  query === '' ||
+                  (queryType === QueryType.Goal && !goal) ||
```

Comment:
The Goal and Neighbour check should be extracted out into a `validQueryTypeInfo` variable or something that sounds better and then used here and with the sync button.

Reply:
Have a look - I added logic to fall back to neighbourhood if there's no goal, so the input can always be made valid, that simplifies the validation and is more robust I think.